library/src-3.x/scala/quoted/matching/ConstSeq.scala

```diff
@@ -0,0 +1,22 @@
+package scala.quoted.matching
+
+import scala.quoted.Expr
+
+import scala.tasty.Reflection // TODO do not depend on reflection directly
+
+/** Literal sequence of literal constant value expressions */
+object ConstSeq {
+
+  /** Matches literal sequence of literal constant value expressions */
+  def unapply[T](expr: Expr[Seq[T]]) given Reflection: Option[Seq[T]] = expr match {
+    case ExprSeq(elems) =>
+      elems.foldRight(Option(List.empty[T])) { (elem, acc) =>
+        (elem, acc) match {
+          case (Const(value), Some(lst)) => Some(value :: lst)
+          case (_, _) => None
+        }
+      }
+    case _ => None
+  }
+
+}
```

Comment:
This looks like to be a util extractor. Why not have a general `SeqLiteral` extractor first?

Reply:
We do have general `SeqLiteral` called `ExprSeq` which is used above. It is true that this is implemented in terms of the other "core" extractors, but it is something that may will use for their first macros.

Reply:
Also, this is in `scala.quoted.matching` which contains all util extractors on `Expr`, not to be confused with the util in `scala.tasty.reflect` which would contain extractors on trees.